src/Microsoft.VisualStudio.Web.CodeGeneration.Utils/DotNet/ResolvedReferenceExtensions.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection.PortableExecutable;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.Web.CodeGeneration.ProjectInfo;
+
+namespace Microsoft.VisualStudio.Web.CodeGeneration.DotNet
+{
+    public static class ResolvedReferenceExtensions
+    {
+        public static IEnumerable<MetadataReference> GetMetadataReference(this ResolvedReference reference, bool throwOnError = true)
+        {
+            var references = new List<MetadataReference>();
+            AssemblyMetadata assemblyMetadata;
+            try
+            {
+                using (var stream = File.OpenRead(reference.ResolvedPath))
+                {
+                    var moduleMetadata = ModuleMetadata.CreateFromStream(stream, PEStreamOptions.PrefetchMetadata);
+                    assemblyMetadata = AssemblyMetadata.Create(moduleMetadata);
+                    references.Add(assemblyMetadata.GetReference());
+                }
+            }
+            catch (Exception ex)
+                when (ex is FileNotFoundException
```

Comment:
Some of these are redundant (e.g. FileNotFoundException inherits from IOException)